<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="adamc" />


<title>Read QC Report</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.5/datatables.js"></script>
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 45px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h2 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h3 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h4 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h5 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h6 {
  padding-top: 50px;
  margin-top: -50px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Analysis of tRNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="QC_alignment.html">read QC</a>
</li>
<li>
  <a href="QC_report.html">
    <span class="fa fa-link"></span>
     
    QC-report
  </a>
</li>
<li>
  <a href="DE.html">
    <span class="fa fa-link"></span>
     
    Diff expression
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Read QC Report</h1>
<h4 class="author">adamc</h4>
<h4 class="date">15 May, 2019</h4>

</div>


<p>Report generated based on data located in /Volumes/Documents2/trna_testing_2/Report  </p>
<p>Samples: test, test_2, test_3, test_4, test_5, test_6 </p>
<p>Report generated at: 08:51:28 15/05/2019 </p>
<div id="general-stats" class="section level1">
<h1>General stats</h1>
<p>MultiQC general stats from Bowtie, featureCounts, FastQC and Samtools. <div id="htmlwidget-47a4cffdb2c396422969" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-47a4cffdb2c396422969">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19"],["cluster","test","test.stats","test_2","test_2.stats","test_2_processed","test_3","test_3.stats","test_3_processed","test_4","test_4.stats","test_4_processed","test_5","test_5.stats","test_5_processed","test_6","test_6.stats","test_6_processed","test_processed"],[99.9,0.03,null,0.03,null,52.93,0.03,null,52.93,0.03,null,52.93,0.03,null,52.93,0.03,null,52.93,52.93],[2996,132,null,132,null,264669,132,null,264669,132,null,264669,132,null,264669,132,null,264669,264669],[null,0.0911072528166327,null,0.0911072528166327,null,null,0.0911072528166327,null,null,0.0911072528166327,null,null,0.0911072528166327,null,null,0.0911072528166327,null,null,null],[null,456,null,456,null,null,456,null,null,456,null,null,456,null,null,456,null,null,null],[null,90.8208,null,90.8208,null,null,90.8208,null,null,90.8208,null,null,90.8208,null,null,90.8208,null,null,null],[null,52,null,52,null,null,52,null,null,52,null,null,52,null,null,52,null,null,null],[null,51,null,51,null,null,51,null,null,51,null,null,51,null,null,51,null,null,null],[null,45.4545454545455,null,45.4545454545455,null,null,45.4545454545455,null,null,45.4545454545455,null,null,45.4545454545455,null,null,45.4545454545455,null,null,null],[null,500000,null,500000,null,null,500000,null,null,500000,null,null,500000,null,null,500000,null,null,null],[null,null,1.948548,null,1.948548,null,null,1.948548,null,null,1.948548,null,null,1.948548,null,null,1.948548,null,null],[null,null,0,null,0,null,null,0,null,null,0,null,null,0,null,null,0,null,null],[null,null,641,null,641,null,null,641,null,null,641,null,null,641,null,null,641,null,null],[null,null,0.128069625121626,null,0.128069625121626,null,null,0.128069625121626,null,null,0.128069625121626,null,null,0.128069625121626,null,null,0.128069625121626,null,null],[null,null,500509,null,500509,null,null,500509,null,null,500509,null,null,500509,null,null,500509,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample Name<\/th>\n      <th>Bowtie: % Reads Aligned<\/th>\n      <th>Bowtie: Reads Aligned<\/th>\n      <th>FC: % Assigned<\/th>\n      <th>FC: Assigned<\/th>\n      <th>FastQC: % Duplicates<\/th>\n      <th>FastQC: % GC<\/th>\n      <th>FastQC: Average Sequence Length<\/th>\n      <th>FastQC: % Fails<\/th>\n      <th>FastQC: Total Sequences<\/th>\n      <th>Samtools: % Error Rate<\/th>\n      <th>Samtools: Non Primary Allignments<\/th>\n      <th>Samtools: Reads Mapped<\/th>\n      <th>Samtools: % Reads Mapped<\/th>\n      <th>Samtools: Raw Total Sequences<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11,12,13,14,15]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script> FC: featureCounts</p>
</div>
<div id="samtools" class="section level1">
<h1>Samtools</h1>
<p>SAM (Sequence Alignment/Map) format is a generic format for storing large nucleotide sequence alignments.</p>
<p>SAM Tools provide various utilities for manipulating alignments in the SAM format, including sorting, merging, indexing and generating alignments in a per-position format (hosted on github: <a href="https://github.com/samtools/samtools" class="uri">https://github.com/samtools/samtools</a>)</p>
<div id="reads-mapped-stats" class="section level2 tabset tabset-pills">
<h2>Reads mapped stats</h2>
<div id="number-of-reads" class="section level3">
<h3>Number of reads</h3>
<p>Mapped vs unmapped reads <img src="QC_report_files/figure-html/samtools_plot1-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="percentages" class="section level3">
<h3>Percentages</h3>
<p>Mapped vs unmapped reads <img src="QC_report_files/figure-html/samtools_plot2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="alignment-metrics" class="section level2">
<h2>Alignment Metrics</h2>
<p>Alignment metrics from samtools stats</p>
<div id="htmlwidget-6c48652e45d43294cc39" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6c48652e45d43294cc39">{"x":{"filter":"none","data":[["1","2","3","4","5","6"],["test","test_2","test_3","test_4","test_5","test_6"],[500509,500509,500509,500509,500509,500509],[641,641,641,641,641,641],[0,0,0,0,0,0],[499868,499868,499868,499868,499868,499868],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0],[32691,32691,32691,32691,32691,32691],[32691,32691,32691,32691,32691,32691],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0],[0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample<\/th>\n      <th>total_sequences<\/th>\n      <th>mapped<\/th>\n      <th>mapped_paired<\/th>\n      <th>unmapped<\/th>\n      <th>properly_paired<\/th>\n      <th>paired<\/th>\n      <th>duplicated<\/th>\n      <th>MQ0_reads<\/th>\n      <th>QC_failed<\/th>\n      <th>bases_mapped<\/th>\n      <th>bases_mapped_(cigar)<\/th>\n      <th>bases_trimmed<\/th>\n      <th>bases_duplicated<\/th>\n      <th>inward_oriented_pairs<\/th>\n      <th>outward_oriented_pairs<\/th>\n      <th>other_orientation<\/th>\n      <th>different_chromosomes<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="mapped-reads-per-contig" class="section level2 tabset tabset-pills">
<h2>Mapped reads per contig</h2>
<p>The samtools idxstats tool counts the number of mapped reads per chromosome / contig. Chromosomes with &lt; 0.1% of the total aligned reads are omitted from this plot</p>
<div id="normalised-counts" class="section level3">
<h3>Normalised counts</h3>
<p><img src="QC_report_files/figure-html/mapped_reads_contig_norm-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="counts" class="section level3">
<h3>Counts</h3>
<p><img src="QC_report_files/figure-html/mapped_reads_count-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="bowtie" class="section level1 tabset tabset-pills">
<h1>Bowtie</h1>
<div id="number-of-reads-1" class="section level2">
<h2>Number of Reads</h2>
<p><img src="QC_report_files/figure-html/number_bowtie_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="percentages-1" class="section level2">
<h2>Percentages</h2>
<p><img src="QC_report_files/figure-html/percent_bowtie_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="feature-counts" class="section level1">
<h1>Feature Counts</h1>
<p>Subread featureCounts is a highly efficient general-purpose read summarization program that counts mapped reads for genomic features. Feature counts obtains data from bam files located in mapping.dir.</p>
<div id="gene-ids" class="section level2 tabset tabset-pills">
<h2>Gene IDs</h2>
<div id="number" class="section level3">
<h3>Number</h3>
<p><img src="QC_report_files/figure-html/fc_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="percent" class="section level3">
<h3>Percent</h3>
<p><img src="QC_report_files/figure-html/fc_plot_percent-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="feature-count-table" class="section level2">
<h2>Feature Count Table</h2>
<div id="htmlwidget-55205eb2c7c0008c88bf" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-55205eb2c7c0008c88bf">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18"],["test_2","test_2","test_2","test_3","test_3","test_3","test_4","test_4","test_4","test_5","test_5","test_5","test_6","test_6","test_6","test","test","test"],["tRNA","srpRNA","rRNA","tRNA","srpRNA","rRNA","tRNA","srpRNA","rRNA","tRNA","srpRNA","rRNA","tRNA","srpRNA","rRNA","tRNA","srpRNA","rRNA"],[7,1,448,7,1,448,7,1,448,7,1,448,7,1,448,7,1,448],[1.53508771929825,0.219298245614035,98.2456140350877,1.53508771929825,0.219298245614035,98.2456140350877,1.53508771929825,0.219298245614035,98.2456140350877,1.53508771929825,0.219298245614035,98.2456140350877,1.53508771929825,0.219298245614035,98.2456140350877,1.53508771929825,0.219298245614035,98.2456140350877]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample<\/th>\n      <th>GeneID<\/th>\n      <th>NumberMapped<\/th>\n      <th>Percentages<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="assigned-unassigned" class="section level2 tabset tabset-pills">
<h2>Assigned/ Unassigned</h2>
<div id="number-of-reads-2" class="section level3">
<h3>Number of reads</h3>
<p><img src="QC_report_files/figure-html/fc_plot2-1.png" width="672" style="display: block; margin: auto;" /> <div id="htmlwidget-71eb007da5f3a3eef6c6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-71eb007da5f3a3eef6c6">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18"],["test","test_2","test_3","test_4","test_5","test_6","test","test_2","test_3","test_4","test_5","test_6","test","test_2","test_3","test_4","test_5","test_6"],["Assigned","Assigned","Assigned","Assigned","Assigned","Assigned","Unassigned_Unmapped","Unassigned_Unmapped","Unassigned_Unmapped","Unassigned_Unmapped","Unassigned_Unmapped","Unassigned_Unmapped","Unassigned_NoFeatures","Unassigned_NoFeatures","Unassigned_NoFeatures","Unassigned_NoFeatures","Unassigned_NoFeatures","Unassigned_NoFeatures"],[456,456,456,456,456,456,499868,499868,499868,499868,499868,499868,185,185,185,185,185,185],[0.0911072528166327,0.0911072528166327,0.0911072528166327,0.0911072528166327,0.0911072528166327,0.0911072528166327,99.8719303748784,99.8719303748784,99.8719303748784,99.8719303748784,99.8719303748784,99.8719303748784,0.0369623723049935,0.0369623723049935,0.0369623723049935,0.0369623723049935,0.0369623723049935,0.0369623723049935]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample<\/th>\n      <th>Status<\/th>\n      <th>Number<\/th>\n      <th>Percentages<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script></p>
</div>
<div id="percentages-2" class="section level3">
<h3>Percentages</h3>
<p><img src="QC_report_files/figure-html/fc_plot3-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="htmlwidget-a3b38e1ef7f8ff67eb00" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a3b38e1ef7f8ff67eb00">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18"],["test","test_2","test_3","test_4","test_5","test_6","test","test_2","test_3","test_4","test_5","test_6","test","test_2","test_3","test_4","test_5","test_6"],["Assigned","Assigned","Assigned","Assigned","Assigned","Assigned","Unassigned_Unmapped","Unassigned_Unmapped","Unassigned_Unmapped","Unassigned_Unmapped","Unassigned_Unmapped","Unassigned_Unmapped","Unassigned_NoFeatures","Unassigned_NoFeatures","Unassigned_NoFeatures","Unassigned_NoFeatures","Unassigned_NoFeatures","Unassigned_NoFeatures"],[456,456,456,456,456,456,499868,499868,499868,499868,499868,499868,185,185,185,185,185,185],[0.0911072528166327,0.0911072528166327,0.0911072528166327,0.0911072528166327,0.0911072528166327,0.0911072528166327,99.8719303748784,99.8719303748784,99.8719303748784,99.8719303748784,99.8719303748784,99.8719303748784,0.0369623723049935,0.0369623723049935,0.0369623723049935,0.0369623723049935,0.0369623723049935,0.0369623723049935]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample<\/th>\n      <th>Status<\/th>\n      <th>Number<\/th>\n      <th>Percentages<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="fast-qc" class="section level1">
<h1>Fast QC</h1>
<p>FastQC is used to perfrom quality control checks on raw high throughput sequencing data.</p>
<div id="mean-quality-scores" class="section level2">
<h2>Mean quality scores</h2>
<p>The mean quality value across each base position in the read. <img src="QC_report_files/figure-html/QC_mean_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="per-sequence-quality-scores" class="section level2">
<h2>Per sequence quality scores</h2>
<p>The number of reads with average quality scores. Shows if a subset of reads has poor quality <img src="QC_report_files/figure-html/per_sequence_qs_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="per-sequence-gc-content" class="section level2">
<h2>Per sequence GC content</h2>
<p>The average GC content of reads. Normal random library typically have a roughly normal distribution of GC content. <img src="QC_report_files/figure-html/per_sequence_gc_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="per-base-n-content" class="section level2">
<h2>Per base N content</h2>
<p>If a sequencer is unable to make a base call with sufficient confidence then it will normally substitute an N rather than a conventional base. This plot shows the percentage of base calls at each position for which an N was called. The documentation states that you get a warning if the number of Ns is above 5%. <img src="QC_report_files/figure-html/per_base_n_count-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sequence-duplicaton-levels" class="section level2">
<h2>Sequence Duplicaton Levels</h2>
<p>The relative level of duplication found for every sequence. <img src="QC_report_files/figure-html/sequence_duplication_levels_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="overrepresented-sequences" class="section level2">
<h2>Overrepresented Sequences</h2>
<p>The total amount of overrepresented sequences found in each library <img src="QC_report_files/figure-html/overrepresented_sequences_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="adapter-content" class="section level2 tabset tabset-pills">
<h2>Adapter content</h2>
<div id="specific-adapter-content" class="section level3">
<h3>Specific adapter content</h3>
<p>The percentage count of the proportion of your library which has seen individual types of adaptor sequences at each position. <img src="QC_report_files/figure-html/adapter_content_plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="adapter-content-cumulative" class="section level3">
<h3>Adapter content cumulative</h3>
<p>The cumulative percentage count of the proportion of your library which has seen each of the adapter sequences at each position. <img src="QC_report_files/figure-html/adapter_content_plot_cumsum-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
